erts/emulator/beam/erl_bits.c

```diff
@@ -1132,17 +1178,21 @@ erts_new_bs_put_float(Process *c_p, Eterm arg, Uint num_bits, int flags)
 #ifdef WORDS_BIGENDIAN
```

Comment:
`bs_construct_SUITE:fp16/1` fails on our big endian machine, can you look into that?

Reply:
Can you please post the failure if handy? I will try to figure out how to run this patch on a big endian machine. :)

Reply:
The first line of the test fails with `{badmatch,false}`, it doesn't say much more :/

I think the lines below are the culprit as they will always be zero when given an fp16, if I'm reading the code right you want to write the first 16 bits instead:

```c
	    t[0] = a >> 24;
	    t[1] = a >> 16;
```

Reply:
Ah, good point, thanks! I believe I have addressed your changes. I kept it as a separate commit for now as it also has some debugging information for the tests. If it works, then I can squash and we can discuss if we should keep the debugging info or no.

Reply:
Thanks, the fix looks a bit strange though: why set `t[2]` and `t[3]` for a 16-bit value? Shouldn't it be `t[0]` and `t[1]`?

```suggestion
#ifdef WORDS_BIGENDIAN
	    if (num_bits == 16) {
	        t[0] = a >> 8;
	        t[1] = a;
	    } else if (num_bits >= 32) {
		t[0] = a >> 24;
		t[1] = a >> 16;
	        t[2] = a >> 8;
	        t[3] = a;
	        
	        if (num_bits == 64) {
		    t[4] = b >> 24;
		    t[5] = b >> 16;
		    t[6] = b >> 8;
		    t[7] = b;
	        }
	    }
```

Reply:
@jhogberg of course you fix makes much more sense. I have applied it to both bigendian branches.

Reply:
Thanks, the tests pass and I'll merge this as soon as I can. There are some bureaucratic hurdles with importing the dependency that I hope will clear soon.